### DISCUSSION TITLE
Best way to make form edition

### DISCUSSION THREAD

**Discussion by ScreamZ** [Akryum/vue-apollo#495] [Q&A]
Hello guys,

I would like to know what's the best way you would recommend me to use in order to make an entity edition form using vue apollo ?

I'm currently using a query that fetch a User object with firstname, lastname and many other fields.
It fills some input fields that display the current value. I would like to edit those values and save changes to dispatch a mutation.

**My problems are:**

- apollo user object is read-only, so i can't modify it.
- I can't initialize my data with the query result because when the component is mounted i can't already access the data…

Is that a good way to deal with this ? Am I forced to use a `watch` and synchronize `user` and `userForm` ?

Best regards

**Comment by andruu** [comment]
I have the same issue and I can't figure out how to solve it. I'm doing a login and if there's an error the email and password fields on the Vue instance become un-reactive.

**Comment by webmint** [comment]
there is 2 ways you can do that. Lets say you receive User object with email && name properties. and u have receivedUser in data.
So u can do `this.receivedUser = JSON.parse(JSON.stringify(user))`
Or u can assign each property separatly.

Since result received from query is always frozen its the only correct ways to do that
